sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -61,8 +70,13 @@ trait HashJoin {
 
       // Mutable per row objects.
       private[this] val joinRow = new JoinedRow2
+      private[this] val resultProjection: Projection = if (supportUnsafe) {
```

Comment:
Within this file, it only looks like `resultProjection` is called from within an `if(supportUnsafe)` branch.  However, I think we do need to guard the creation of this `UnsafeProjection` behind this `if` to make sure that we don't try to generate it in the non-Unsafe case.  Therefore, this seems fine to me; the `else` branch will never get hit but I suppose it's better to have a safe alternative there instead of `null`.